src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1672,36 +1715,74 @@ void UpdateNavigationBarVisibility(bool animated)
 
 			void UpdateToolbarItems()
 			{
-				if (NavigationItem.RightBarButtonItems != null)
+				// Unsubscribe from previous toolbar item property changes
+				foreach (var item in _trackedToolbarItems)
+				{
+					item.PropertyChanged -= OnToolbarItemPropertyChanged;
+				}
+				_trackedToolbarItems.Clear();
+
+				if (NavigationItem.RightBarButtonItems is not null)
 				{
 					for (var i = 0; i < NavigationItem.RightBarButtonItems.Length; i++)
+					{
 						NavigationItem.RightBarButtonItems[i].Dispose();
+					}
 				}
-				if (ToolbarItems != null)
+
+				if (ToolbarItems is not null)
 				{
 					for (var i = 0; i < ToolbarItems.Length; i++)
+					{
 						ToolbarItems[i].Dispose();
+					}
 				}
 
 				List<UIBarButtonItem> primaries = null;
-				List<UIBarButtonItem> secondaries = null;
+				List<UIMenuElement> secondaries = null;
 				var toolbarItems = _tracker.ToolbarItems;
+				
+				// Subscribe to property changes for all current toolbar items
 				foreach (var item in toolbarItems)
 				{
+					item.PropertyChanged += OnToolbarItemPropertyChanged;
+					_trackedToolbarItems.Add(item);
+
 					if (item.Order == ToolbarItemOrder.Secondary)
-						(secondaries = secondaries ?? new List<UIBarButtonItem>()).Add(item.ToUIBarButtonItem(true));
+					{
+						(secondaries ??= []).Add(item.ToSecondaryToolbarItem().PlatformAction);
+					}
 					else
-						(primaries = primaries ?? new List<UIBarButtonItem>()).Add(item.ToUIBarButtonItem());
+					{
+						(primaries ??= []).Add(item.ToUIBarButtonItem());
+					}
 				}
 
-				if (primaries != null)
+				if (primaries is not null && primaries.Count > 0)
+				{
 					primaries.Reverse();
-				NavigationItem.SetRightBarButtonItems(primaries == null ? Array.Empty<UIBarButtonItem>() : primaries.ToArray(), false);
-				ToolbarItems = secondaries == null ? Array.Empty<UIBarButtonItem>() : secondaries.ToArray();
+				}
 
-				NavigationRenderer n;
-				if (_navigation.TryGetTarget(out n))
+				if (secondaries is not null && secondaries.Count > 0)
+				{
+					var menu = UIMenu.Create(string.Empty, null, UIMenuIdentifier.Edit, UIMenuOptions.DisplayInline, secondaries.ToArray());
+					// Using the ellipsis.circle system image for the menu button, as per the iOS design guidelines: https://developer.apple.com/design/human-interface-guidelines/pull-down-buttons
+					var menuButton = new UIBarButtonItem(UIImage.GetSystemImage("ellipsis.circle"), menu);
```

Comment:
should we make this in a way user can override ? maybe a virtual method on the renderer ?

Reply:
Yep, good idea, done!